src/swap-helpers.ts

```diff
@@ -267,3 +278,23 @@ export const isLikeKind = (
   }
   return false
 }
+
+export const getTokenId = (
+  coreWallet: EdgeCurrencyWallet,
+  currencyCode: string
+): string | undefined => {
+  const { allTokens } = coreWallet.currencyConfig
+  return Object.keys(allTokens).find(
+    edgeToken => allTokens[edgeToken].currencyCode === currencyCode
+  )
+}
+
+export const consify = (val: any): void =>
+  console.log(JSON.stringify(val, null, 2))
+
+export const makeQueryParams = (params: JsonObject): string => {
+  const cleaned = asQueryParams(params)
+  return Object.entries(cleaned)
+    .map(([key, value]) => `${key}=${value.toString()}`)
+    .join('&')
+}
```

Comment:
We should not be cleaning stuff our own program produces. Cleaners are for "outside" data sources like disk or network. A better solution is to delete the cleaner and actually type what we mean:

```ts
export interface QueryParams {
  [key: string]: number | string | boolean | undefined | null
}
```

You should also handle the `null` case by skipping the `=${value}` bit, since sometimes you can put raw keys into query strings like `?count=1&sorted&descending`. Maybe `.map(([key, value]) => value == null ? key : '${key}=${encodeURIComponent(value)}')`, to also handle escaping.

It's also worth noting that `Object.entries` is not going to work on older phones, so those aren't going to support DEX swaps on Thorchain (TypeError: Object.entries is not a function). You could use `Object.keys`, which is much older, and then grab out the value using [], or we could just be fine with the failed quotes.

Reply:
Problem is typescript doesn't like a fixed object passed into a function that accepts a map. This is the object being created

```
      const quoteParams: ThorSwapQuoteParams = {
        sellAsset:
          `${fromMainnetCode}.${fromCurrencyCode}` + (fromTokenId ?? ''),
        buyAsset: `${toMainnetCode}.${toCurrencyCode}` + (toTokenId ?? ''),
        sellAmount,
        slippage: (volatilitySpreadFinal * 100).toString(),
        recipientAddress: toAddress,
        senderAddress: fromAddress,
        affiliateAddress: thorname,
        affiliateBasisPoints: affiliateFeeBasis
      }
      ```
      
      Once this is passed into `makeQueryParams` which accepts a map, TS complains that
      
      ```
      Argument of type 'ThorSwapQuoteParams' is not assignable to parameter of type 'QueryParams'.
  Index signature for type 'string' is missing in type 'ThorSwapQuoteParams'.ts(2345)
```

Reply:
Sam figured out a hack to get this to work as a type instead of cleaner. Basically `ThorSwapQuoteParams` needs to be a `type` not an `interface` then it can be cast into a map. I updated the fixup commit with this change